webapp/src/Controller/PublicController.php

```diff
@@ -116,6 +95,114 @@ public function scoreboardAction(Request $request): Response
+        $basePathProperty  = $requestReflection->getProperty('basePath');
+        $basePathProperty->setValue($request, '/CHANGE_ME');
+        $requestStack->push($request);
+
+        $contestPage = $this->renderView('public/scoreboard.html.twig', $data);
+
+        // Now get all assets that are used
+        $assetRegex = '|/CHANGE_ME/([/a-z0-9_\-\.]*)(\??[/a-z0-9_\-\.=]*)|i';
+        preg_match_all($assetRegex, $contestPage, $assetMatches);
+        $contestPage = preg_replace($assetRegex, '$1$2', $contestPage);
+
+        $zip = new ZipArchive();
+        if (!($tempFilename = tempnam($this->dj->getDomjudgeTmpDir(), "contest-"))) {
+            throw new ServiceUnavailableHttpException(null, 'Could not create temporary file.');
+        }
+
+        $res = $zip->open($tempFilename, ZipArchive::OVERWRITE);
+        if ($res !== true) {
+            throw new ServiceUnavailableHttpException(null, 'Could not create temporary zip file.');
+        }
+        $zip->addFromString('index.html', $contestPage);
+
+        $publicPath = realpath(sprintf('%s/public/', $projectDir));
+        foreach ($assetMatches[1] as $file) {
+            $zip->addFile($publicPath . '/' . $file, $file);
+        }
+
+        // Also copy in the webfonts
+        $webfontsPath = $publicPath . '/webfonts/';
+        foreach (glob($webfontsPath . '*') as $fontFile) {
+            $fontName = basename($fontFile);
+            $zip->addFile($fontFile, 'webfonts/' . $fontName);
+        }
+
+        $zip->close();
+
+        $zipFilename = 'contest.zip';
+
+        $response = new StreamedResponse();
+        $response->setCallback(function () use ($tempFilename) {
+            $fp = fopen($tempFilename, 'rb');
+            fpassthru($fp);
+            unlink($tempFilename);
+        });
+        $response->headers->set('Content-Type', 'application/zip');
+        $response->headers->set('Content-Disposition', 'attachment; filename="' . $zipFilename . '"');
+        $response->headers->set('Content-Length', filesize($tempFilename));
+        $response->headers->set('Content-Transfer-Encoding', 'binary');
+        $response->headers->set('Connection', 'Keep-Alive');
+        $response->headers->set('Accept-Ranges', 'bytes');
+
+        return $response;
+    }
+
+    /**
+     * Get the contest from the request, if any
+     */
+    protected function getContestFromRequest(Request $request): ?Contest
+    {
+        $contest = null;
+        // For static scoreboards, allow to pass a contest= param.
+        if ($contestId = $request->query->get('contest')) {
+            if ($contestId === 'auto') {
+                // Automatically detect the contest that is activated the latest.
+                $contest      = null;
+                $activateTime = null;
+                foreach ($this->dj->getCurrentContests(-1) as $possibleContest) {
+                    if (!$possibleContest->getPublic() || !$possibleContest->getEnabled()) {
```

Comment:
```suggestion
                    if (!($possibleContest->getPublic() && $possibleContest->getEnabled())) {
```
I find this easier to read (I think you just copied this from what was there already)